LLVM 3.8 fails to build on OS X 10.11

## Issue description
Seems like it's just a hash mismatch, but I'm not certain.

```
~ > nix-env -i llvm
installing ‘llvm-3.8.0’
these derivations will be built:
  /nix/store/08fy4bggcvgck44jsds0kzaa3md5n1x2-libcxxabi-3.8.0.src.tar.xz.drv
  /nix/store/fyz4jqrccjqfl3cm5n2if4bvdl6k3c9i-libcxx-3.8.0.src.tar.xz.drv
  /nix/store/k04fx9qdz339dy3hpf0k4vpsqw8m0gp0-libc++abi-3.8.0.drv
  /nix/store/jjnfrdz0cj91zdixssbshbbd3wnj9z81-llvm-3.8.0.drv
building path(s) ‘/nix/store/qzdqpri06k1im9kdl3gr80ii5dksmbsb-libcxx-3.8.0.src.tar.xz’
trying http://llvm.org/releases/3.8.0/libcxx-3.8.0.src.tar.xz
output path ‘/nix/store/qzdqpri06k1im9kdl3gr80ii5dksmbsb-libcxx-3.8.0.src.tar.xz’ has sha256 hash ‘0yr3fh8vj38289b9cwk37zsy7x98dcd3kjy7xxy8mg20p48lb01n’ when ‘0i7iyzk024krda5spfpfi8jksh83yp3bxqkal0xp76ffi11bszrm’ was expected
cannot build derivation ‘/nix/store/k04fx9qdz339dy3hpf0k4vpsqw8m0gp0-libc++abi-3.8.0.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/jjnfrdz0cj91zdixssbshbbd3wnj9z81-llvm-3.8.0.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/jjnfrdz0cj91zdixssbshbbd3wnj9z81-llvm-3.8.0.drv’ failed
```

### Steps to reproduce

`nix-env -i llvm-3.8.0`

## Technical details

* System: OS X 10.11
* Nix version: nix-env (Nix) 1.11.2

* Nixpkgs version: Given command doesn't work, likely due to my configuration, but nix-channel --update downloads from nixpkgs-16.09pre80183.551296a

Alright, after fixing the hash issue, both libcxx and libc++abi needed to be changed, I ran into another issue. When building sanitizer_mac.cc in compiler-rt, the compiler spits something like this out:
```
/private/var/folders/gn/bmrb7z1x36x57qvfl77myt140000gn/T/nix-build-llvm-3.8.0.drv-0/llvm/projects/co
mpiler-rt/lib/sanitizer_common/sanitizer_mac.cc:446:55: error: too few arguments provided to functio
n-like macro invocation
        os_trace("Address Sanitizer reported a failure.");

/nix/store/5rcaaig3p888kagqx66l0mran7f2j88r-Libsystem-osx-10.9.5/include/os/trace.h:46:9: note: macro 'os_trace' defined here
#define os_trace(tag, fmt, ...) __extension__({\

/private/var/folders/gn/bmrb7z1x36x57qvfl77myt140000gn/T/nix-build-llvm-3.8.0.drv-0/llvm/projects/co
mpiler-rt/lib/sanitizer_common/sanitizer_mac.cc:446:7: error: use of undeclared identifier 'os_trace
'; did you mean '_os_trace'?
      os_trace("Address Sanitizer reported a failure.");
      ^~~~~~~~
      _os_trace
/nix/store/5rcaaig3p888kagqx66l0mran7f2j88r-Libsystem-osx-10.9.5/include/os/trace.h:35:1: note: '_os
_trace' declared here                                                                              
_os_trace(const char *msg, ...);
```
Since the 3.8 nix expressions were basically copy-pasted from 3.7, as shown by the hash error, and llvm 3.7 evidently built fine for OS X, this is strange. I'll try building everything outside of nix and see what happens.

--- a/pkgs/development/compilers/llvm/3.8/libc++/default.nix
+++ b/pkgs/development/compilers/llvm/3.8/libc++/default.nix
@@ -3,7 +3,7 @@
 stdenv.mkDerivation rec {
   name = "libc++-${version}";
 
-  src = fetch "libcxx" "0i7iyzk024krda5spfpfi8jksh83yp3bxqkal0xp76ffi11bszrm";
+  src = fetch "libcxx" "0yr3fh8vj38289b9cwk37zsy7x98dcd3kjy7xxy8mg20p48lb01n";
 
   postUnpack = ''
     unpackFile ${libcxxabi.src}

--- a/pkgs/development/compilers/llvm/3.8/libc++abi.nix
+++ b/pkgs/development/compilers/llvm/3.8/libc++abi.nix
@@ -3,7 +3,7 @@
 stdenv.mkDerivation {
   name = "libc++abi-${version}";
 
-  src = fetch "libcxxabi" "0ambfcmr2nh88hx000xb7yjm9lsqjjz49w5mlf6dlxzmj3nslzx4";
+  src = fetch "libcxxabi" "0175rv2ynkklbg96kpw13iwhnzyrlw3r12f4h09p9v7nmxqhivn5";
 
   buildInputs = [ cmake ] ++ stdenv.lib.optional (!stdenv.isDarwin && !stdenv.isFreeBSD) libunwind;
 

--- a/pkgs/development/compilers/llvm/3.8/llvm.nix
+++ b/pkgs/development/compilers/llvm/3.8/llvm.nix
@@ -35,6 +35,13 @@ in stdenv.mkDerivation rec {
 
   propagatedBuildInputs = [ ncurses zlib ];
 
+  # hacky fix: New LLVM releases require a newer OS X SDK than
+  # 10.9. This is a temporary measure until nixpkgs darwin support is
+  # updated.
+  patchPhase = stdenv.lib.optionalString stdenv.isDarwin ''
+        sed -i 's/os_trace(\(.*\)");$/printf(\1\\n");/g' ./projects/compiler-rt/lib/sanitizer_common/sanitizer_mac.cc
+  '';
+
   # hacky fix: created binaries need to be run before installation
   preBuild = ''
     mkdir -p $out/

